lms/djangoapps/edraak_certificates/utils.py

```diff
@@ -0,0 +1,211 @@
+from bidi.algorithm import get_display
+from django.conf import settings
+from django.core.cache import cache
+import os
+import re
+
+from courseware.access import has_access
+from lms.djangoapps.courseware.views.views import is_course_passed
+from opaque_keys.edx import locator
+from xmodule.modulestore.django import modulestore
+
+from edraak_certificates import arabic_reshaper
+
+
+STATIC_DIR = os.path.join(os.path.dirname(__file__), 'assets')
+
+
+def cached_function(cache_key_format, timeout=30):
+    """
+    Decorator to cache heavy functions.
+    Use it as the following:
+    @cached_function("module.add_numbers.{0}.{1}", 30)
+    def add_numbers(a, b):
+        return a + b
+    """
+    def the_decorator(func):
+
+        def cached_func(*args, **kwargs):
+            cache_key = cache_key_format.format(*args, **kwargs)
+            cached_result = cache.get(cache_key)
+
+            if cached_result is not None:
+                return cached_result
+            else:
+                result = func(*args, **kwargs)
+                cache.set(cache_key, result, timeout)
+                return result
+
+        return cached_func
+
+    return the_decorator
+
+
+def is_certificates_feature_enabled():
+    if not settings.FEATURES.get('EDRAAK_CERTIFICATES_APP'):
+        return False
+
+    if not settings.FEATURES.get('ORGANIZATIONS_APP'):
+        raise Exception('Please enable the organization app (`ORGANIZATIONS_APP`) in '
```

Comment:
Can we change the exception message here to a more descriptive one like you enabled X without enabling Y.